config/src/main/java/org/springframework/security/config/annotation/web/configurers/saml2/Saml2LoginConfigurer.java

```diff
@@ -211,7 +225,10 @@ public void configure(B http) throws Exception {
 	}
 
 	private AuthenticationProvider getAuthenticationProvider() {
-		AuthenticationProvider provider = new OpenSamlAuthenticationProvider();
+		OpenSamlAuthenticationProvider provider = new OpenSamlAuthenticationProvider();
+		if (this.userDetailsService != null) {
+			provider.setUserDetailsService(this.userDetailsService);
+		}
```

Comment:
I like that you can explicitly set it. In addition to this, what if there is a 
```
@Bean 
UserDetailsService
``` 
in the context. We should pick up that bean too

Reply:
Hello Fillip,
thank you for your review!

I like your idea of automatically using the UserDetailsService from @Bean registry, if this is present.

However my implementation of manually setting (or maybe overriding) to a specific UserDetailsService was explicitly intended by me:
This way you would be able to have two different UserDetailsServices at the same time and handle SAML authentication in a different way. For example, you could explicitly create a new User as a sideeffect, if that one was not known in a local database. The other UserDetailsService could be used to only authenticate already known users an authenticate by username+password.

Maybe we could combine your and my idea:
User the spring @Bean UserDetailsService as default and also be able to overrride it with the method I already included (setUserDetailsService)?